src/core/EventBus.js

```diff
@@ -49,11 +51,13 @@ function EventBus() {
 
         var handler = {
             callback: listener,
-            scope: scope
+            scope: scope,
+            priority: priority !== undefined ? priority : LOW_PRIORITY
         };
 
         handlers[type] = handlers[type] || [];
         handlers[type].push(handler);
+        handlers[type].sort(function (a, b) {return a.priority - b.priority;});
```

Comment:
Fair point. I'm not saying this isn't a good idea, just that it might not be the final solution. Ship it!